### DISCUSSION TITLE
ASGI Lifespan errored in Litestar

### DISCUSSION THREAD

**Discussion by winstxnhdw** [emmett-framework/granian#356] [Q&A] [answered]
I am migrating my Litestar app from Uvicorn to Granian but I am getting,

```python
[WARNING] ASGI Lifespan errored, continuing without Lifespan support (to avoid Lifespan completely use "asginl" interface)
```

Seeing #297, it seems this warning is issued if the framework doesn't support Lifespans, but I am certain that Litestar does.

**Comment by gi0baro** [comment, marked as answer]
I should probably review the logging for lifespan, as if the application won't return a lifespan error but just throw an exception, then Granian consider it as *unsupported* and avoid to log the actual exception – several people complained by log pollution in the past.

There's definitely something going on in the litestar app during lifespan startup, but In order to retrieve the error, you should copy the ASGI spawn method from here https://github.com/emmett-framework/granian/blob/master/granian/server.py#L216-L275, edit the line

```python
if lifespan_handler.interrupt:
```

and change it to

```python
if lifespan_handler.exc:
```

So you can pass it to the `Granian.serve` method as follows:

```python
Granian(...).serve(spawn_target=your_method)
```

I know this is quite ugly and very far from being ideal, but at the time being is the only option to log that exception.
Probably I can change to code to always log that exception in debug level or something similar.
